### PR TITLE
Jeremy lig 741 improve datasource docs

### PR DESCRIPTION
**Pull request by japrescott**
closes lig-741
closes lig-871
- highlight important "click" parts of the datasource documentations
- correct statement that streaming is only possible with S3 and link to GCS and Azure docs
- add note that instance and bucket should be in the same region

### CODE REVIEW

File: docs/source/docker/integration/docker_with_datasource.rst [hunk 1]
```diff
@@ -16,8 +16,13 @@ the following workloads in one single run:
...
 
-Support for streaming from Google Cloud Storage and Azure Blob Storage is
-planned. If you need any of these, write us so that we prioritize implementing it.
+Streaming from Google Cloud Storage and Azure Blob Storage is also supported.
+Please follow the
+`Google Cloud Storage <https://docs.lightly.ai/getting_started/dataset_creation/dataset_creation_gcloud_bucket.html`_
+ or the
+`Azure Blob Storage <https://docs.lightly.ai/getting_started/dataset_creation/dataset_creation_azure_storage.html>`_
+ guide to learn how to setup your bucket and configure a datasource.
+
```
Comment (guarin) [inline docs/source/docker/integration/docker_with_datasource.rst:25]: GCS and Azure support is not yet implemented in the docker :(  There seems to be an issue for M0.1 but it is not in the cycle: https://linear.app/lightly/issue/LIG-770/make-compute-worker-docker-work-with-azure-datasource

File: docs/source/docker/integration/docker_with_datasource.rst [hunk 2]
```diff
@@ -34,6 +39,13 @@ Advantages
   maintaining your privacy and security.
 
 
+.. note:: Please ensure that the region of your bucket and where you intend to be running the
+          Lightly Docker instance should be the same (e.g. `eu-central-1`). If the region is not
+          the same there can be
+          `degraded transfer speeds and additional costs will be incurred by AWS <https://aws.amazon.com/premiumsupport/knowledge-center/s3-transfer-data-bucket-instance/>`_!
+          We highly recommend using the same region
+
```
Comment (guarin) [inline docs/source/docker/integration/docker_with_datasource.rst:47]: `.` missing at end of sentence.

File: docs/source/getting_started/dataset_creation/dataset_creation_azure_storage.rst [hunk 3]
```diff
@@ -21,8 +21,8 @@ For the purpose of this guide we assume you have a storage account called `light
 We further assume the container you want to use with lightly is called `farm-animals` and already contains images.
 If you don't have a storage account or container yet follow the instructions `here <https://docs.microsoft.com/en-us/azure/storage/common/storage-account-create?tabs=azure-portal>`_.
 
-Go to "Security + networking > Shared access signature". There, configure the access rights to your liking. Lightly requires at least read and list access to the container. Make sure to
-also tick the boxes for the required resource types (Container and Object). Set an expiry date and then click on "Generate SAS and connection string". Copy the generated "SAS token" including the "?"
+Go to **"Security + networking"** > **"Shared access signature"**. There, configure the access rights to your liking. Lightly requires at least read and list access to the container. Make sure to
```
Comment (guarin) [inline docs/source/getting_started/dataset_creation/dataset_creation_azure_storage.rst:24]: Maybe add a note that write access is required for thumbnails, videos and object level stuff.

File: docs/source/getting_started/dataset_creation/dataset_creation_azure_storage.rst [hunk 4]
```diff
@@ -32,8 +32,8 @@ Preparing your data
...
  
 1. Install AzCopy cli by following the `guide of Azure <https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10>`_
-2. To sync your data from the container to your local machine (or vice versa), you need a shared access token. On your *storage account* page of the Azure portal, on the left, go to "Security + networking > Shared access signature". Generate a shared access signature (SAS) which allows access to the container and objects.
-3. Copy the SAS token and use the following command to sync the Azure blob storage with your local folder:
+2. To sync your data from the container to your local machine (or vice versa), you need a shared access token. On your `storage account`` page of the Azure portal, on the left, go to **"Security + networking"** > **"Shared access signature"**. Generate a shared access signature (SAS) which allows access to the container and objects.
+3. Copy the `SAS token`` and use the following command to sync the Azure blob storage with your local folder:
```
Comment (guarin) [inline docs/source/getting_started/dataset_creation/dataset_creation_azure_storage.rst:36]: There seem to be some single ` and some double ``. I guess all of them should be single?

File: docs/source/getting_started/dataset_creation/dataset_creation_azure_storage.rst [hunk 5]
```diff
@@ -54,7 +54,7 @@ Uploading your data
         :align: center
         :alt: Edit your dataset.
 
-        To edit your dataset click the `Edit` button on the top.
+        To edit your dataset click the **`Edit`** button on the top.
```
Comment (guarin) [inline docs/source/getting_started/dataset_creation/dataset_creation_azure_storage.rst:57]: The Edit is surrounded by `` and not by ""

**Review by guarin — CHANGES_REQUESTED** [review]
Looks good :) There are some minor formatting mistakes and we sadly don't yet fully support GCS and Azure...

File: docs/source/docker/integration/docker_with_datasource.rst (hunk 1 above)
Comment (japrescott) [reply docs/source/docker/integration/docker_with_datasource.rst:25]: ach :(

**Comment by japrescott** [comment]
thanks @guarin for your review. I've addressed your remarks

**Review by guarin — APPROVED** [review]
One double `` is still in there, otherwise looks good!
I'll approve already :)

File: docs/source/getting_started/dataset_creation/dataset_creation_azure_storage.rst [hunk 6]
```diff
@@ -32,7 +33,7 @@ Preparing your data
 For the :ref:`lightly-command-line-tool` to be able to create embeddings and extract metadata from your data, `lightly-magic` needs to be able to access your data. You can download/sync your data from Azure blob storage.
  
 1. Install AzCopy cli by following the `guide of Azure <https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10>`_
-2. To sync your data from the container to your local machine (or vice versa), you need a shared access token. On your `storage account`` page of the Azure portal, on the left, go to **"Security + networking"** > **"Shared access signature"**. Generate a shared access signature (SAS) which allows access to the container and objects.
+2. To sync your data from the container to your local machine (or vice versa), you need a shared access token. On your `storage account` page of the Azure portal, on the left, go to **"Security + networking"** > **"Shared access signature"**. Generate a shared access signature (SAS) which allows access to the container and objects.
 3. Copy the `SAS token`` and use the following command to sync the Azure blob storage with your local folder:
```
[the remark this answers is not in the document]
Comment (japrescott) [reply docs/source/getting_started/dataset_creation/dataset_creation_azure_storage.rst]: sorry about that

**Pull request merged**